### PR TITLE
initial investigation to create a travis.ci build

### PR DESCRIPTION
**Pull request by DanHeidinga**
Initial investigation to get a travis.ci build up and running as per #24.

Commits will be squashed and cleaned up before merging.  PR is just so others can follow along (or make suggestions on)  the approach being taken.

### CODE REVIEW

**Comment by mgaudet** [comment]
Regarding the re-clone thing -- If you know what dir Travis has cloned into, i have previously successfully used `get_sources.sh` with a local directory; it means two copies on disk, yes, but the clone is nearly instantaneous.

**Comment by DanHeidinga** [comment]
@mgaudet Travis supplies the build directory in one of the env vars:
```
TRAVIS_BUILD_DIR: The absolute path to the directory where the repository being built has been copied on the worker.
```
Do you have a command line you can share that shows the local clone?

**Comment by mgaudet** [comment]
So, I wasn't working on travis, so this is a small shot in the dark, but I was thinking this

```
 bash ./get_source.sh -openj9-repo=$TRAVIS_BUILD_DIR -openj9-branch=$TRAVIS_BRANCH -openj9-sha=$TRAVIS_COMMIT
```

Assuming travis doesn't do something silly on the clone (like nuke the `.git` dir), it should be fine to just provide the filesystem path as the argument to `get_sources.sh`

**Comment by DanHeidinga** [comment]
Thanks @mgaudet!  That worked much better than my earlier attempts to craft a file:// url.

**Comment by DanHeidinga** [comment]
Reclone now works.  Next challenge: boot idk not being recognized.

**Comment by DanHeidinga** [comment]
Boot JDK is recognized and supported now.
Had to disable ccache due to the openjdk configure script rejecting the wrapped compiler.

Current status: build failed due to g++ internal compiler error.  Relaunching

**Comment by DanHeidinga** [comment]
First successful compile attempt in the travis env using 1 core for both the configure and for the make commands. 
Next step: try to increase that and see if the builds are faster while still passing.

**Comment by DanHeidinga** [comment]
Complete build and 'java -version' runs in ~30 min on travis.ci without ccache enabled.

**Comment by DanHeidinga** [comment]
I think this is ready to go as an initial travis build.  It only ensures the code compiles and can run -version but we can look at enabling testing (and ccache) in separate PRs.

**Comment by DanHeidinga** [comment]
@mgaudet Can you review this and call out any issues?  Once its approved, the one of the other committers can also review and merge.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+script:
+  # Clear this option so it doesn't interfere with configure detecting the bootjdk
+  - unset _JAVA_OPTIONS
+  # Point the get_sources script at the OpenJ9 repo that's already been cloned to disk.  
+  # Results in a copy of the source (disk space =( ) but no new network activity so overall a win. 
+  - cd openj9-openjdk-jdk9 && bash ./get_source.sh -openj9-repo=$TRAVIS_BUILD_DIR -openj9-branch=$TRAVIS_BRANCH -openj9-sha=$TRAVIS_COMMIT
+  # Based on https://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/ travis container builds
+  # have 2 cores and 4 gigs of memory.  Attempt to double provision the number of cores for the make...
+  # Limit number of jobs to work around g++ internal compiler error
+  - export UMA_WINDOWS_PARRALLEL_HACK="-j4"
```
Comment (mgaudet) [inline .travis.yml]: Is the hack actually misspelled? (Two `R`s)

**Comment by mgaudet** [comment]
To be clear, in the final merged version you do, or do not want ccache? (I know the OpenJDK build scripts demand `--with-ccache` ) 

Also, what's the distinction between `make all` and `make images`?

File: .travis.yml (hunk 1 above)
Comment (DanHeidinga) [reply .travis.yml]: Sadly, that matches what's in the tr.source makefiles: https://github.com/eclipse/openj9/blob/master/runtime/tr.source/makefile.ftl 

(edited to add: and in the rest of our code... see https://github.com/eclipse/openj9/search?utf8=✓&q=UMA_WINDOWS_PARRALLEL_HACK&type=)

Comment (mgaudet) [reply .travis.yml]: facepalm

**Comment by mgaudet** [comment]
Ok -- with that in mind, I think this is fine to land as is, with the potential to iterate 

👍 :shipit:

**Comment by DanHeidinga** [comment]
Can one of the committers approve / merge this?  @charliegracie @pshipton others?

**Pull request merged**